code/clue.js

```diff
@@ -1,59 +1,260 @@
+  material: 'silver'
+}
+
+const pistol = {
+  name: 'Pistol',
+  material: 'metal'
 }
 
 // THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+
+// STEP 2 - CREATE ARRAYS TO SUSPECTS, WEAPONS AND ROOMS 
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
+  professorPlum,
+  colonelMustard,
+  mrsPeacock,
+  mrsWhite,
+  missScarlett
+]
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
+
+const rooms = [
+  "Dining Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio"
 ]
 
-const weapons = []
+// STEP 3 - PICK A MURDERER, A WEAPON AND A ROOM
 
-const rooms = []
+// OBJECT THAT KEEPS THE MYSTERY (with a killer, a weapon and a room)
+
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null
+}
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
 
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
   // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
+  document.getElementById('killerText').style.background = mystery.killer.color
   document.getElementById(
-    'killerName'
+    'killerText'
   ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerImage').style.background = mystery.killer.image
+}
+
+// This function will be invoked when you click on the weapon card.
+const pickWeapon = () => {
+
+  mystery.weapon = randomSelector(weapons)
+
+  document.getElementById(
+    'weaponText'
+  ).innerHTML = `${mystery.weapon.name}`
+}
+
+// This function will be invoked when you click on the room card.
+const pickRoom = () => {
+
+  mystery.room = randomSelector(rooms)
+
+  document.getElementById(
+    'roomText'
+  ).innerHTML = `${mystery.room}`
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+document.getElementById('killer').addEventListener('click', pickKiller)
+document.getElementById('weapon').addEventListener('click', pickWeapon)
+document.getElementById('room').addEventListener('click', pickRoom)
```

Comment:
Glad to see all of your listeners in the same place.  Would like to see reveal here as well.